silx/gui/plot/StatsWidget.py

```diff
@@ -955,19 +1039,30 @@ def __init__(self, parent=None, plot=None, stats=None):
         qt.QWidget.__init__(self, parent)
         self.setLayout(qt.QVBoxLayout())
         self.layout().setContentsMargins(0, 0, 0, 0)
-        self._options = _OptionsWidget(parent=self)
+        self._options = _OptionsWidget(parent=self, updateMode=UpdateMode.manual)
         self.layout().addWidget(self._options)
         self._statsTable = StatsTable(parent=self, plot=plot)
+        self._statsTable.setDisplayOnlyActiveItem(self._options.isActiveItemMode())
+        self._options._setUpdateMode(mode=self._statsTable.getUpdateMode())
         self.setStats(stats)
 
         self.layout().addWidget(self._statsTable)
 
+        old = self._statsTable.blockSignals(True)
         self._options.itemSelection.triggered.connect(
             self._optSelectionChanged)
         self._options.dataRangeSelection.triggered.connect(
             self._optDataRangeChanged)
-        self._optSelectionChanged()
         self._optDataRangeChanged()
+        self._statsTable.blockSignals(old)
+
+        self._statsTable.sigUpdateModeChanged.connect(self._options._setUpdateMode)
+        callback = functools.partial(self._getStatsTable()._updateAllStats, is_request=True)
+        self._options.sigUpdateStats.connect(callback)
+
+        # expose API
+        self.getUpdateMode = self._statsTable.getUpdateMode
+        self.setUpdateMode = self._statsTable.setUpdateMode
```

Comment:
Maybe better to have proxy methods, it is more explicit. (There is a `@docstring` decorator in `silx.utils.proxy` to duplicate the docstrings in proxy).